### ISSUE TITLE
[unbound-method] Add an option to ignore annotated methods

### ISSUE THREAD

**Issue by vlsergey** [typescript-eslint/typescript-eslint#3245]
- [ ✓ ] I have tried restarting my IDE and the issue persists.
- [ ✓ ] I have updated to the latest version of the packages.
- [ ✓ ] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": 2,
  }
}
```

```TS
import React, { PureComponent, ReactNode } from 'react'
import { boundMethod } from 'autobind-decorator';

class MyComponent extends PureComponent<{}> {

    @boundMethod
    handleChange( event : React.ChangeEvent<HTMLInputEvent> ) {
        this.setState( { value: event.currentTarget.value } );
    }

    render() : ReactNode {
        return <input onChange={this.handleChange} />
    }
}
```

```json
{
  "compilerOptions": {
    "declaration": true,
    "declarationMap": true,
    "esModuleInterop": true,
    "experimentalDecorators": true,
    "jsx": "react",
    "lib": ["dom", "es5", "es6"],
    "moduleResolution": "node",
    "noImplicitAny": true
  }
}
```

**Expected Result**

* I would like to have an option of unbound-method rule to ignore methods, annotated by any/some/specific annotations.

**Actual Result**

* No such option present
* Error is reported

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.19.0` |
| `@typescript-eslint/parser`        | `4.19.0` |
| `TypeScript`                       | `4.2.3` |
| `ESLint`                           | `7.22.0` |
| `node`                             | `12.18.0` |

**Comment by bradzacher** [comment]
why not just use class properties with arrow functions which are auto-bound by default and do not require you to bring in the decorator codegen?

```ts
import React, { PureComponent, ReactNode } from 'react'
import { boundMethod } from 'autobind-decorator';

class MyComponent extends PureComponent<{}> {

    handleChange = ( event : React.ChangeEvent<HTMLInputEvent> ) => {
        this.setState( { value: event.currentTarget.value } );
    }

    render() : ReactNode {
        return <input onChange={this.handleChange} />
    }
}
```

**Comment by vlsergey** [comment]
Looks like possible solution for this particular case. Well, can be closed for now, but i assume more cases can occur in future.
